inst/scripts/generate-table-ad.R

```diff
@@ -0,0 +1,372 @@
+          annots = annots,
+          success = "false",
+          task_view = opts$task_view
+        )
+      }
+      failure_message <- glue::glue(
+        "Could not gather task annotations:\n  {e$message}"
+      )
+      error(logger, failure_message)
+      quit(status = 1)
+    }
+  )
+}
+
+## Setup done ------------------------------------------------------------------
+
+## Get study metadata file IDs -----
+all_files <- tryCatch(
+  {
+    purrr::map_dfr(opts$directories, ~ gather_metadata_synIDs_all(.))
+  },
+  error = function(e) {
+    if (update_task) {
+      update_task_annotation(
+        task_id = opts$task_id,
+        annots = annots,
+        success = "false",
+        task_view = opts$task_view
+      )
+    }
+    failure_message <- glue::glue(
+      "There was a problem getting synIDs for metadata files:\n  {e$message}"
+    )
+    error(logger, failure_message)
+    quit(status = 1)
+  }
+)
+
+## Remove files that are probably not metadata -----
+
+## Only keep csv files that don't have 'dictionary' in name
+all_files <- all_files[grepl(".csv$", all_files$name, ignore.case = TRUE), ]
+all_files <- all_files[!grepl("dictionary", all_files$name, ignore.case = TRUE), ] # nolint
+## Fix id column to be character
+all_files$id <- as.character(all_files$id)
+
+## Grab file view for annotations -----
+
+view_query <- tryCatch(
+  {
+    synapser::synTableQuery(
+      glue::glue("SELECT * FROM {opts$file_view}")
+    )$asDataFrame()
+  },
+  error = function(e) {
+    if (update_task) {
+      update_task_annotation(
+        task_id = opts$task_id,
+        annots = annots,
+        success = "false",
+        task_view = opts$task_view
+      )
+    }
+    failure_message <- glue::glue(
+      "There was a problem getting the file view:\n  {e$message}"
+    )
+    error(logger, failure_message)
+    quit(status = 1)
+  }
+)
+
+## Check that view has needed columns
+
+missing_cols <- setdiff(
+  unique(c(FILE_VIEW_COLUMNS_BIND, FILE_VIEW_COLUMNS_JOIN)),
+  colnames(view_query)
+)
+if (length(missing_cols) > 0) {
+  if (update_task) {
+    update_task_annotation(
+      task_id = opts$task_id,
+      annots = annots,
+      success = "false",
+      task_view = opts$task_view
+    )
+  }
+  missing <- glue::glue_collapse(missing_cols, sep = ", ")
+  failure_message <- glue::glue(
+    "The file view is missing these columns:\n  {missing}"
+  )
+  error(logger, failure_message)
+  quit(status = 1)
+}
+
+## Grab metadata-related annotations from file view and join -----
+
+view_meta <- view_query[view_query$id %in% all_files$id, ]
+all_files <- dplyr::left_join(
+  all_files,
+  view_meta[, FILE_VIEW_COLUMNS_JOIN]
+)
+## Remove any with metadataType dictionary or protocol
+all_files <- all_files[!grepl("dictionary|protocol", all_files$metadataType), ]
+
+## Join metadata -----
+
+## Open files and gather IDs
+all_meta_ids <- tryCatch({
+    gather_ids_all_studies(all_files) -> all_meta_id
```

Comment:
```suggestion
    gather_ids_all_studies(all_files) -> all_meta_ids
```
Not sure if this is supposed to generate two separate data frames but they were identical and I didn't see `all_meta_id` used anywhere else in the script.

Reply:
😆 You found a mistake! I used that when I was testing and forgot to remove. 🤦🏻